js/bundle_test.go

```diff
@@ -588,6 +588,10 @@ func TestOpen(t *testing.T) {
 
 func TestBundleInstantiate(t *testing.T) {
 	b, err := getSimpleBundle("/script.js", `
+		export let options = {
```

Comment:
Not sure this is the best test for this feature. It doesn't hurt to have it, but it's not testing that options that are injected from outside in the JS runtime. In the real k6 execution, since CLI flags and environment variables have precedence over exported script `options`, once we consolidate them, we write them back in the JS runtimes, so the script can read their actual values: https://github.com/loadimpact/k6/blob/16b2a1ba2373c5a63d8a533eef788389c944c69c/cmd/run.go#L184-L187

Something like that isn't tested here

Reply:
@na-- I meant the original test as a unit test since the change was in `Bundle.Instantiate()`, but then left it after the move to `ForEachSpecified` as an "indirect" unit test, but mostly out of laziness :smile:

I can add a higher level functional test for this, but I'm wondering if there's a need for it. If I'm interpreting `cmd/run.go` correctly, `getConsolidatedConfig()` is called on line 139, and both the Runner and the Engine receive the consolidated options, so by the time `Bundle.Instantiate()` is called the config hierarchy has already been resolved and we wouldn't need to test this resolution for this change.

Correct me if I'm wrong, and if so, what would be the best place for this higher level test (should we test at the Bundle or Runner level)?

Reply:
No need to go much higher level than this, you just need to test the proposition that changing the options in the `Bundle` will then be reflected in the actual `options` object in the running script. After all, that's what `SetOptions()` does: https://github.com/loadimpact/k6/blob/731e55be472895ffac6d5d77c218ffade98453e8/js/runner.go#L275

So, you should be able to do something like this at the end of this same test:
```
b.Options.VUs = 55 // or whatever, something to simulate call to the runner's SetOptions()
bi2, err := b.Instantiate() // make a new instance
// verify that bi2's `options` contains the externally updated value, not the one initially specified in the script
```
```

Reply:
Thanks, I misunderstood your original comment. The test should be fixed now.